Lib/fnmatch.py

```diff
@@ -61,6 +61,22 @@ def filter(names, pat):
                 result.append(name)
     return result
 
+def filterfalse(names, pat):
+    """Construct a list from those elements of the iterable NAMES that do not match PAT."""
+    result = []
+    pat = os.path.normcase(pat)
+    match = _compile_pattern(pat)
+    if os.path is posixpath:
```

Comment:
```py
if os.path is not posixpath:
    names = map(os.path.normcase, names)
return itertools.filterfalse(match, names)
```

The only difference would be usage of `filter` instead of `filterfalse` in normal case.

Also, `itertools` is already in dependency path.

Would be interesting to see if this performs better.

Reply:
I'll run the benchmarks. I wanted to avoid importing itertools but if it's already cached maybe it's fine.

Reply:
For posterity: itertools is imported because of: `functools -> reprlib -> itertools`

Reply:
IIRC, `itertools` also is basically free when it comes to import time by virtue of being written in C

Reply:
On small iterables *names*, the loop is better (which was what I expected) but for larger data, it's `itertools.filterfalse`, when you match a lot of patterns or not. So I may transition to the `itertools` solution.

Reply:
Yes, but for posixpath I think it's sufficiently good. I'm not sure if using map would be faster though, I'll benchmark it.

Reply:
Err.... actually, I don't have access to a Windows machine... so I can't benchmark it...

Reply:
Yeah, but that is the point, you can not use map, because `os.path.normcase` is not applied to result.

I have experimenting with `pipe` concept for a while now. I think it is useful, but still unsure if it is worth proposal. Essentially what you would need:

```py
return itertools.filterfalse(pipe(os.path.normcase, match), names)
```

Currently, the only option is to use lambda:
```py
return itertools.filterfalse(lambda n: match(os.path.normcase(n)), names)
```
but this is very likely to be too slow.

If you are going to try it, predefine `normcase`:
```py
normcase = os.path.normcase
return itertools.filterfalse(lambda n: match(normcase(n)), names)
```

To avoid double attribute access on each iteration.

Reply:
> Err.... actually, I don't have access to a Windows machine... so I can't benchmark it...

You don't need access to windows machine to benchmark this. This is only because on posix there is no case sensitivity and normalization only needs to be done on windows. Benchmark for this will be valid on non-windows machine too.

Reply:
Apologies, mistakes in my previous comment.

I will put it a bit differently. Although functionality on windows is different, benchmark on non-windows machine will be valid, because the point of optimization is function composition and not the performance of `normcase`.

In other words, if it will be faster on posix, it will be faster on windows too, because `normcase` is a constant overhead and its performance has no effect on any of the other parts of construct.